modules/command/common/dataset-pruning-command.js

```diff
@@ -52,22 +48,56 @@ class DatasetPruningCommand extends Command {
                 offerIdToBeDeleted,
                 dataInfoIdToBeDeleted,
                 datasetsToBeDeleted,
+                bidIdToBeDeleted,
+                numberOfPrunedDatasets,
             } = response;
-            if (datasetsToBeDeleted.length === 0) {
-                return;
-            }
+
             await this.datasetPruningService.removeEntriesWithId('offers', offerIdToBeDeleted);
             await this.datasetPruningService.removeEntriesWithId('data_info', dataInfoIdToBeDeleted);
-
+            await this.datasetPruningService.removeEntriesWithId('bids', bidIdToBeDeleted);
             await this.datasetPruningService.updatePruningHistory(datasetsToBeDeleted);
-            this.logger.info(`Sucessfully pruned ${datasetsToBeDeleted.length} datasets.`);
+
+            if (this.datasetPruningService.shouldPruneLowEstimatedValueDatasets()) {
+                const datasets = await this.datasetPruningService.findLowEstimatedValueDatasets();
+
+                if (!datasets) {
+                    await this.addPruningCommandToExecutor();
+                    return;
+                }
+
+                const repackedDatasets = this.datasetPruningService
+                    .repackLowEstimatedValueDatasets(datasets);
+
+                forked.send(JSON.stringify({
+                    selectedDatabase: this.config.database,
+                    lowEstimatedValueDatasetsPruning: true,
+                    repackedDatasets,
+                    numberOfPrunedDatasets,
+                }));
+                return;
+            }
+            if (numberOfPrunedDatasets > 0) {
+                this.logger.info(`Successfully pruned ${numberOfPrunedDatasets} datasets.`);
+            } else {
+                this.logger.info('Found 0 datasets for pruning');
+            }
             forked.kill();
             await this.addPruningCommandToExecutor();
         });
         this.logger.trace('Dataset pruning worker started');
         return Command.empty();
     }
 
+    /**
+     * Recover system from failure
+     * @param command
+     * @param err
+     */
+    async recover(command, err) {
```

Comment:
Where is this function called?

Reply:
This method is automatically called if command fails for any unhandled exception

Reply:
Thanks